src/main/java/org/broadinstitute/hellbender/tools/exome/SegmentUtils.java

```diff
@@ -13,292 +15,452 @@
+        for (final String contig : breakpointsByContig.keySet()) {
+            final List<Breakpoint> breakpoints = breakpointsByContig.get(contig);
+            int start = breakpoints.get(0).getStart();
+            for (int i = 1; i < breakpoints.size(); i++) {
+                final Breakpoint breakpoint = breakpoints.get(i);
+                int end;
+                //adjust segment boundaries according to breakpoint type; this prevents, e.g., SNPs that originally
+                //started SNP segments from being stranded
+                switch (breakpoint.getType()) {
+                    case START: end = breakpoint.getSite() - 1;
+                        break;
+                    case END: end = breakpoint.getSite();
+                        break;
+                    default: throw new GATKException.ShouldNeverReachHereException("Unknown breakpoint type in segment union.");
+                }
 
-        final double segMean = segment.getSegmentMeanInCRSpace();
-        final List<TargetCoverage> myTargets = targets.targets(segment);
-        return myTargets.stream().map(t -> ( Math.pow(2, t.getCoverage()) - segMean)).collect(Collectors.toList());
+                if (end < start) {  //this could happen if there are adjacent breakpoints
+                    continue;
+                }
+
+                final SimpleInterval segment = new SimpleInterval(contig, start, end);
+                if (genome.getTargets().targetCount(segment) > 0 || genome.getSNPs().targetCount(segment) > 0) {
+                    segments.add(segment);
+                }
+                start = segment.getEnd() + 1;
+            }
+        }
+        return segments;
     }
 
-    /**
-     * write a list of intervals with calls to file.
-     *
-     * Writes segment mean in CR space, not log2CR.
-     */
-    public static void writeModeledSegmentsToSegfile(final File outFile, List<ModeledSegment> segments, final String sample) {
-        try (final TableWriter<ModeledSegment> writer = TableUtils.writer(outFile,
-                new TableColumnCollection(SegmentTableColumns.COLUMN_NAME_ARRAY),
-
-                //lambda for filling an initially empty DataLine
-                (ms, dataLine) -> {
-                    dataLine.append(sample, ms.getContig()).append(ms.getStart(), ms.getEnd())
-                            .append(ms.getOriginalProbeCount())
-                            .append(ms.getSegmentMeanInCRSpace())
-                            .append(ms.getCall());
-                })) {
-            for (final ModeledSegment ms : segments) {
-                writer.writeRecord(Utils.nonNull(ms, "Segments list contains a null."));
+    //remerge spurious target-only segments introduced at starts and ends of the
+    //original target-coverage segments by union of breakpoints
+    private static List<SimpleInterval> mergeSpuriousStartsAndEnds(final List<SimpleInterval> targetSegments,
```

Comment:
`segments` should be the first argument because it contains the segments whose spurious starts and ends are being merged.